### PR TITLE
feat(chat): implement Client Data Service: save/read/update installed deployments into ClientData (Issue #2034)

### PR DESCRIPTION
**Pull request by IlyaBondar**
**Description:**

implement Client Data Service: save/read/update installed deployments into ClientData

Issues:

- Issue #2034

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by ai-dial-actions** [comment]
Hey there and thank you for opening this pull request! 👋🏼

We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.

Details:

```
No subject found in pull request title "feat(chat): ".
```

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2120.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2120.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/constants/client-data.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const CLIENTDATA_PATH = 'clientdata';
+export const INSTALLED_DEPLOYMENTS = 'installed_deployments.json';
```
Comment (Alexander-Kezik) [inline apps/chat/src/constants/client-data.ts:2]: I think these variables are destined to be in the config, what do you think about that @IlyaBondar?

File: apps/chat/src/store/models/models.epics.ts [hunk 2]
```diff
@@ -129,6 +132,51 @@ const getModelsEpic: AppEpic = (action$, state$) =>
     }),
   );
 
+const getInstalledModelIdsEpic: AppEpic = (action$, state$) =>
+  action$.pipe(
+    filter(ModelsActions.getInstalledModelIds.match),
+    withLatestFrom(state$),
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/models/models.epics.ts]: do we need `withLatestFrom`?

File: apps/chat/src/store/models/models.epics.ts [hunk 3]
```diff
@@ -129,6 +132,51 @@ const getModelsEpic: AppEpic = (action$, state$) =>
...
+          return of(ModelsActions.getInstalledModelIdsFail());
+        }),
+      );
+    }),
+  );
+
+const getInstalledModelIdsFailEpic: AppEpic = (action$, state$) =>
+  action$.pipe(
+    filter(ModelsActions.getInstalledModelIdsFail.match),
+    withLatestFrom(state$),
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/models/models.epics.ts]: do we need `withLatestFrom`?

File: apps/chat/src/store/models/models.epics.ts [hunk 4]
```diff
@@ -129,6 +132,51 @@ const getModelsEpic: AppEpic = (action$, state$) =>
...
+    switchMap(() => {
+      const defaultModelIds = SettingsSelectors.selectDefaultRecentModelsIds(
+        state$.value,
+      );
+      return of(ModelsActions.updateInstalledModelIds(defaultModelIds));
+    }),
+  );
+
+const updateInstalledModelIdsEpic: AppEpic = (action$) =>
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/models/models.epics.ts:164]:
```suggestion
const updateInstalledDeploymentIdsEpic: AppEpic = (action$) =>
```

I think we should use or Deployments or models

File: apps/chat/src/constants/client-data.ts (hunk 1 above)
Comment (IlyaBondar) [reply apps/chat/src/constants/client-data.ts:2]: no, it shouldn't be configured

File: apps/chat/src/store/models/models.epics.ts (hunk 3 above)
Comment (IlyaBondar) [reply apps/chat/src/store/models/models.epics.ts]: $state is used later here

File: apps/chat/src/store/models/models.epics.ts (hunk 4 above)
Comment (IlyaBondar) [reply apps/chat/src/store/models/models.epics.ts:164]: In our code we are using term `models` (all neighboring fields)

File: apps/chat/src/store/models/models.epics.ts (hunk 3 above)
Comment (Alexander-Kezik) [reply apps/chat/src/store/models/models.epics.ts]: But we already have it as argument

**Review by Alexander-Kezik — APPROVED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2120.nightly-test.deltixhub.io
>E2E tests status: failed

**Pull request merged**